superset/assets/src/chart/Chart.jsx

```diff
@@ -124,6 +137,17 @@ class Chart extends React.PureComponent {
       return this.renderStackTraceMessage();
     }
 
+    const showOverlay = formData.show_overlay;
+
+    if (showOverlay && formData.hasOwnProperty('extra_filters') && formData['extra_filters'].length > 0) {
+      formData.show_overlay = false;
+      showOverlay = false;
```

Comment:
should not we use formData.show_overlay instead of local variable?

Reply:
This has been discussed earlier, `form_data` would be updating multiple times in render function so had to kept the local variable to get the right state.